### PR TITLE
Java examples sql

### PR DESCRIPTION
**Pull request by adam47deg**
Changes for 'Expose Java wrapper for integrations/sql' ticket:
https://app.clickup.com/t/861n05tth

New files
  DatabaseExample.java - Java clone of the Kotlin DatabaseExample.kt
  AIDatabase.java - wrapper for accessing JDBCSQLImpl, SQL methods in Java
  ExecutionContext.java - holder of shared executor/futures and CoreAIScope
  Util.java - Java console read line util that works in your IDE

Updated files
  AIScope.java - Refactored to use new ExecutionContext.java for shared coreScope and future exec
  PDFDocument.java - Update to use Util.java to read from console
  DatabaseExample.kt - Made console output on a single line to improve readability

### CODE REVIEW

File: examples/java/src/main/java/com/xebia/functional/xef/java/auto/util/Util.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package com.xebia.functional.xef.java.auto.util;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+public class Util {
+
+    private static final BufferedReader sysin = new BufferedReader(new InputStreamReader(System.in));
```
Comment (Zevleg) [inline examples/java/src/main/java/com/xebia/functional/xef/java/auto/util/Util.java:9]: Both BufferedReader and InputStreamReader implement AutoCloseable and should be specified in try-with-resources blocks so they can be properly finalized and closed when done.

File: integrations/sql/src/main/kotlin/com/xebia/functional/xef/sql/SQL.kt [hunk 2]
```diff
@@ -15,9 +15,13 @@ import kotlinx.serialization.json.Json
 interface SQL {
 
   companion object {
-    suspend fun <A> fromJdbcConfig(config: JdbcConfig, block: suspend SQL.() -> A): A = JDBCSQLImpl(config).use {
+    suspend fun <A> fromJdbcConfig(config: JdbcConfig, block: suspend SQL.() -> A) : A = JDBCSQLImpl(config).use {
       block(it)
     }
+
+    fun fromJdbcConfigSync(config: JdbcConfig): SQL {
```
Comment (Zevleg) [inline integrations/sql/src/main/kotlin/com/xebia/functional/xef/sql/SQL.kt:22]: Function "fromJdbcConfigSync" is never used

File: java/src/main/java/com/xebia/functional/xef/java/auto/ExecutionContext.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    private final CoreAIScope scope;
+
+    public ExecutionContext(){
+        this(Executors.newCachedThreadPool(new ExecutionContext.AIScopeThreadFactory()),  new OpenAIEmbeddings(OpenAI.DEFAULT_EMBEDDING));
+    }
+
+    public ExecutionContext(ExecutorService executorService, Embeddings embeddings) {
+        this.executorService = executorService;
+        this.coroutineScope = () -> ExecutorsKt.from(executorService).plus(JobKt.Job(null));
+        JakartaValidationModule module = new JakartaValidationModule(
```
Comment (Zevleg) [inline java/src/main/java/com/xebia/functional/xef/java/auto/ExecutionContext.java:36]: Variable 'module' is never used

Comment (adam47deg) [reply java/src/main/java/com/xebia/functional/xef/java/auto/ExecutionContext.java]: Fixed

File: integrations/sql/src/main/kotlin/com/xebia/functional/xef/sql/SQL.kt (hunk 2 above)
Comment (adam47deg) [reply integrations/sql/src/main/kotlin/com/xebia/functional/xef/sql/SQL.kt:22]: This is being used in AIDatabase.java

File: examples/java/src/main/java/com/xebia/functional/xef/java/auto/util/Util.java (hunk 1 above)
Comment (adam47deg) [reply examples/java/src/main/java/com/xebia/functional/xef/java/auto/util/Util.java]: Fixed. Added Autoclosable interface and renamed to ConsoleUtil.java

**Review by Zevleg — APPROVED** [review]

**Pull request merged**